[CP] Migrate command usage values (#139383)

eliasyishak opened the pull request:
Fixes:
- https://github.com/flutter/flutter/issues/141017

christopherfujino committed:
[flutter_tools] disable flutter build AAR for plugins (#137878)

Per discord discussion, building an AAR out of a plugin project has not worked for years, so let's just disable the functionality. Context: https://github.com/flutter/flutter/issues/137564

eliasyishak committed:
Migrate command usage values (#139383)

Related to the tracker issue:
- https://github.com/flutter/flutter/issues/128251

This PR migrates the `Usage.command` static method that sent custom dimensions for each command (if applicable). The screenshot below shows the different places where the `usageValues` getter is overwritten to return the necessary custom dimensions for that command.

<img width="285" alt="image" src="https://github.com/flutter/flutter/assets/42216813/e32d5100-0e17-4a4d-8f21-327a8c113a19">

eliasyishak committed:
Importing analytics packages in files

eliasyishak committed:
Remove `artifacts` param from build command for test

eliasyishak committed:
Add required imports for `common.dart` in `test`

eliasyishak committed:
`flutter update-packages --cherry-pick-package=dwds --cherry-pick-version='21.0.0+1'`

christopherfujino commented on packages/flutter_tools/lib/src/android/gradle.dart:
```diff
@@ -640,10 +640,9 @@ class AndroidGradleBuilder implements AndroidBuilder {
-    if (!manifest.isModule && !manifest.isPlugin) {
-      throwToolExit('AARs can only be built for plugin or module projects.');
+    if (!manifest.isModule) {
+      throwToolExit('AARs can only be built for module projects.');
```
is this related?

eliasyishak replied:
No, this was brought in from your [commit](https://github.com/flutter/flutter/pull/141019/commits/d2bd4441c0c6330a6729c0a5c37b3a68a4efdb50) actually. There was a change you made that my CP was depending on. I thought originally that I would need to only merge previously committed PRs for a CP.

It's probably best to revert the changes from your commit and just make the necessary changes for this PR to work right?

christopherfujino commented on packages/flutter_tools/lib/src/commands/build_aar.dart:
```diff
@@ -102,6 +123,14 @@ class BuildAarCommand extends BuildSubCommand {
       'This cannot currently be configured.';
 
+  @override
+  Future<void> validateCommand() async {
```
same question, is this required for usage values?

eliasyishak replied:
It is not required for this PR

eliasyishak replied:
Well turns out this was necessary for this [test](https://github.com/eliasyishak/flutter/blob/cp-migrate-usage-values-ga4/packages/flutter_tools/test/commands.shard/hermetic/build_aar_test.dart#L70-L73) 🙃

christopherfujino replied:
Let's delete the test rather than change the behavior of the tool in CP.

eliasyishak committed:
Revert "[flutter_tools] disable flutter build AAR for plugins (#137878)"

This reverts commit d2bd4441c0c6330a6729c0a5c37b3a68a4efdb50.

eliasyishak committed:
Use getter for `FlutterProject`

eliasyishak committed:
Reland the `validateCommand` method for build_aar.dart

eliasyishak committed:
Update build_aar_test.dart

eliasyishak committed:
Update logic for getting `Analytics` instance from package:unified_analytics (#134756)

Part of:
- https://github.com/flutter/flutter/issues/128251

Currently, when we want to use the analytics instance from `package:unified_analytics`, we are just grabbing it from globals. However, with the legacy analytics instance, there are some things we check to return a no-op version of the instance.. for example, if we are running on bots or a non standard branch, we use a no-op instance

This PR uses the same previous checks for the new analytics instance

eliasyishak committed:
Update build_aar_test.dart

eliasyishak commented:
@christopherfujino looks like this is ready for your review now

christopherfujino commented on packages/flutter_tools/test/commands.shard/hermetic/build_aar_test.dart:
```diff
@@ -0,0 +1,136 @@
+    );
+  });
+
+  testUsingContext('will not build an AAR for a plugin', () async {
```
If the tested behavior did not already exist on the 3.16 branch, let's just delete this test, per https://github.com/flutter/flutter/pull/141019/files#r1445425246

christopherfujino reviewed (approved):
This LGTM once you remove the `BuildAarCommand.validateCommand()` override and the failing test.

eliasyishak committed:
Remove aar related code + test file

eliasyishak committed:
Merge branch 'flutter-3.16-candidate.0' into cp-migrate-usage-values-ga4

eliasyishak committed:
Empty commit to kick presubmits
